### PR TITLE
quality shit and removing domains name from source

### PR DESCRIPTION
**Pull request by ImZaw**

### CODE REVIEW

**Comment by ImZaw** [comment]
wait dont push

**Comment by C10udburst** [comment]
> wait dont push

bruhh u know that draft PRs exist right?

**Comment by ImZaw** [comment]
> > wait dont push
> 
> bruhh u know that draft PRs exist right?

what is this

**Comment by C10udburst** [comment]
draft prs prevents us from merging unless u undraft them

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request

**Comment by ImZaw** [comment]
> draft prs prevents us from merging unless u undraft them
> 
> https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request

well i was gonna push a push rq and merge

**Comment by ImZaw** [comment]
anywYs i think u can merge

**Review by C10udburst — APPROVED** [review]

File: NineGoalProvider/src/main/kotlin/com/ninegoal/9GoalProvider.kt [hunk 1]
```diff
@@ -115,22 +116,31 @@ class NineGoal : MainAPI() {
...
         sourcesData?.playUrls?.apmap {
+            val quality = it.name?.substringAfter("(")?.substringBefore(")").let {
+                when (it) {
+                    "Full HD" -> 1080
+                    "HD" -> 720
+                    "SD" -> 480
+                    else -> Qualities.Unknown.value
+                }
+            }
+            val language = it.name?.replace(" (${it.name?.substringAfter("(")?.substringBefore(")")})", "") ?: ""
```
Comment (Blatzar) [inline NineGoalProvider/src/main/kotlin/com/ninegoal/9GoalProvider.kt:127]: regex here please

**Review by Blatzar — CHANGES_REQUESTED** [review]

File: NineGoalProvider/src/main/kotlin/com/ninegoal/9GoalProvider.kt (hunk 1 above)
Comment (ImZaw) [reply NineGoalProvider/src/main/kotlin/com/ninegoal/9GoalProvider.kt]: “ \(.*“
should it be double \\ or im not sure cause im on phone

Comment (ImZaw) [reply NineGoalProvider/src/main/kotlin/com/ninegoal/9GoalProvider.kt]: before the (

File: NineGoalProvider/src/main/kotlin/com/ninegoal/9GoalProvider.kt [hunk 2]
```diff
@@ -58,6 +59,11 @@ data class sourceData (
 data class sourcesJSON (
     @JsonProperty("data" ) var data : sourceData? = sourceData()
 )
+
+private fun String.getDomainFromUrl(): String? {
+    return Regex("""^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n\?\=]+)""").find(this)?.groupValues?.firstOrNull()
```
Comment (Blatzar) [inline NineGoalProvider/src/main/kotlin/com/ninegoal/9GoalProvider.kt:64]: Whats happening here?

**Pull request merged**